internal/serve/serve.go

```diff
@@ -207,17 +208,25 @@ func Serve(opts ServeOptions, httpServer HTTPServerInterface) error {
 	return nil
 }
 
+const (
+	rateLimitPer20Seconds = 30
+	rateLimitWindow       = 20 * time.Second
+)
+
 func handleHTTP(o ServeOptions) *chi.Mux {
 	mux := chi.NewMux()
 
 	// Middleware
 	mux.Use(middleware.CorsMiddleware(o.CorsAllowedOrigins))
 	mux.Use(chimiddleware.RequestID)
 	mux.Use(chimiddleware.RealIP)
+	mux.Use(httprate.LimitAll(rateLimitPer20Seconds, rateLimitWindow))
```

Comment:
Is this 90 rpm for all IPs or 90rpm per IP ?

Reply:
Looking at the [source code](https://github.com/go-chi/httprate/blob/master/httprate.go#L17-L19) looks like it's for all IPs. I'd suggest we use the following configuration:

```suggestion
	mux.Use(httprate.Limit(
		rateLimitPer20Seconds,
		rateLimitWindow,
		httprate.WithKeyFuncs(httprate.KeyByIP, httprate.KeyByEndpoint),
	))
```

Reply:
Good catch, TY!

Reply:
I followed Caio's suggestion in https://github.com/stellar/stellar-disbursement-platform-backend/pull/195/commits/303a904b5738efc729ad107d411ca544f6a504ea, and updated the limit per window.

The new limit is 10 requests every 20 seconds with the pair {IP, endpoint}. Which translates to a maximum of 30 requests per minute per IP, segregated per endpoint.